FParsecCS/Buffer.cs

```diff
@@ -23,188 +25,38 @@ public static class Buffer {
-                *((int*)dst) = *((int*)src);
-            }
-            if ((size & 2) != 0) {
-                src -= 2; dst -= 2;
-                *((short*)dst) = *((short*)src);
-            }
-            if ((size & 1) != 0) {
-                src -= 1; dst -= 1;
-                *dst = *src;
-            }
-        }
-        return;
-    } else {
-        // backup path for pointers with different byte (or 2-byte) alignment
-        for (; size != 0; --size) {
-            --src; --dst;
-            *dst = *src;
-        }
-        return;
-    }
+    System.Buffer.MemoryCopy(src, dst, size, size);
 }
 
 #endif
 
-internal static uint SwapByteOrder(uint value) {
-    return    (((value << 24) | (value >>  8)) & 0xff00ff00U)
-            | (((value <<  8) | (value >> 24)) & 0x00ff00ffU);
-}
+internal static uint SwapByteOrder(uint value) => BinaryPrimitives.ReverseEndianness(value);
 
-internal static ulong SwapByteOrder(ulong value) {
-    return   (((value << 56) | (value >>  8)) & 0xff000000ff000000UL)
-           | (((value <<  8) | (value >> 56)) & 0x000000ff000000ffUL)
-           | (((value << 40) | (value >> 24)) & 0x00ff000000ff0000UL)
-           | (((value << 24) | (value >> 40)) & 0x0000ff000000ff00UL);
-}
+internal static ulong SwapByteOrder(ulong value) => BinaryPrimitives.ReverseEndianness(value);
 
-internal static void SwapByteOrder(uint[] array) {
+internal static void SwapByteOrder(Span<uint> array) {
     for (int i = 0; i < array.Length; ++i) {
-        var v = array[i];
-        array[i] =   (((v << 24) | (v >>  8)) & 0xff00ff00U)
-                   | (((v <<  8) | (v >> 24)) & 0x00ff00ffU);
-    }
-}
-
-#if !LOW_TRUST
-
-internal static unsafe void SwapByteOrder(uint* buffer, uint length) {
-    for (int i = 0; i < length; ++i) {
-        var v = buffer[i];
-        buffer[i] =   (((v << 24) | (v >>  8)) & 0xff00ff00U)
-                    | (((v <<  8) | (v >> 24)) & 0x00ff00ffU);
+        array[i] = BinaryPrimitives.ReverseEndianness(array[i]);
     }
 }
 
-#endif
-
 #if LOW_TRUST
-
-internal static byte[] CopySubarray(byte[] array, int index, int length) {
-    var subArray = new byte[length];
-    System.Buffer.BlockCopy(array, index, subArray, 0, length);
-    return subArray;
-}
-
-internal static uint[] CopyUIntsStoredInLittleEndianByteArray(byte[] src, int srcIndex, int srcLength) {
+internal static uint[] CopyUIntsStoredInLittleEndianByteArray(ReadOnlySpan<byte> src, int srcIndex, int srcLength) {
     Debug.Assert(srcLength%sizeof(uint) == 0);
     var subArray = new uint[srcLength/sizeof(uint)];
-    System.Buffer.BlockCopy(src, srcIndex, subArray, 0, srcLength);
+    src.Slice(srcIndex, srcLength).CopyTo(MemoryMarshal.AsBytes(new Span<uint>(subArray)));
     if (!BitConverter.IsLittleEndian) SwapByteOrder(subArray);
     return subArray;
 }
-
 #endif
 
 #if !LOW_TRUST
 // used by StaticMapping.createStaticStringMapping
 public static unsafe bool Equals(uint* ptr1, uint* ptr2, uint length) {
```

Comment:
I could make this use `MemoryExtensions.SequenceEqual` if the length was an `int`…

Reply:
Should be safe for string comparisons. But you'd have to touch `StaticMapping.fs` 😁

Reply:
Turns out the length is a compile-time constant and always signed. Done.